### ISSUE TITLE
Getting rid of patched pack_fopen

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#383]
A reason why AGS needs to patch Allegro 4's file.c is the use of overwritten pack_fopen (also mentioned in the discussion to #346).

I finally got to investigate why this functions is required at all, and this is what I found:
1) The function is only used when loading audio files from VOX or main game file.
2) The function corrects datafile lookup in some way more convenient for AGS. That includes opening stream from the package of custom format.
3) Some of the audio types are being loaded through the use of Allegro's functions, which take only filename as parameter, but in turn call pack_fopen internally, such as [load_midi](https://github.com/adventuregamestudio/ags/blob/master/Engine/media/audio/sound.cpp#L266), for example. Because of this, there is no clear way to make them create audio streams from custom file location.

**NOTE:** By the way, the VOX in AGS are basically same data container as the main game file, albeit of older format version. They are created in the Editor's native code, function called [make_old_style_data_file](https://github.com/adventuregamestudio/ags/blob/master/Editor/AGS.Native/agsnative.cpp#L2147). The code there might need refactoring, but in general it seem to create same format as former clib32 unit, now rewritten as the AssetManager class. In fact, AssetManager successfully reads the table of contents from those files, as can be seen e.g. in engine's function [UpgradeAudio](https://github.com/adventuregamestudio/ags/blob/master/Common/game/main_game_file.cpp#L517).

**TO SUMMARIZE**, in order to get rid of this function, we need to figure out how to create Allegro's audio streams from an arbitrary memory buffer or file stream, which AGS loads itself from whichever place it needs, OR how to make Allegro's function pack_fopen get file stream from some kind of callback AGS provides.

**PS**. This ofcourse won't be needed if AGS audio system would be overwirtten to use something else instead of Allegro 4.

**Comment by ivan-mogilko** [comment]
Upon further investigation, there is a general issue with Allegro 4 that its API does not provide functions that take PACK_FILE pointer to load certain objects, but only filenames, so you cannot do custom stream opening. "load_midi" is one example of that.

By the way, I just noticed that Allegro 4.4 branch has active commits again, maybe there is a chance we may propose such changes to them, or even make pull request.

**Comment by ivan-mogilko** [comment]
There is an old ticket regarding this problem in Allegro's project: https://github.com/liballeg/allegro5/issues/460

So far, I think load_midi is the only actual problem. Other sound and video types seem to provide means to create object from passed stream, but I need to do some experiments first to see if that is possible to cover all cases.

**Comment by ivan-mogilko** [comment]
This issue has been addressed in "release-3.4.1" and "master" branches, where pack_fopen hack was replaced.

Additional functions which load MIDI sound and FLIC from arbitrary PACKFILE stream were contributed to Allegro 4 library:
https://github.com/liballeg/allegro5/pull/729
https://github.com/liballeg/allegro5/pull/761
